### PR TITLE
[Agent] Fix deadlock and empty rx counter and rx_bytes counter 

### PR DESCRIPTION
**Pull request by TomatoMr**
### This PR is for:

- Agent

#### Steps to reproduce the bug
- 
#### Changes to fix the bug
- 
#### Affected branches
- main
#### Checklist
- [ ] Added unit test to verify the fix.

### CODE REVIEW

**Review by rvql — APPROVED** [review]

File: agent/src/trident.rs [hunk 1]
```diff
@@ -485,7 +485,19 @@ fn dispatcher_listener_callback(
                 );
                 listener.on_vm_change(&vm_mac_addrs);
             }
-            components.tap_typer.on_tap_types_change(tap_types); // FIXME: it is necessary to judge whether the tap_types are updated
+            let mut updated = true;
+            if components.cur_tap_types.len() != tap_types.len() {
```
Comment (yuanchaoa) [inline agent/src/trident.rs:489]: 仅用长度判断是否更新不准确吧

Comment (yuanchaoa) [reply agent/src/trident.rs:489]: !=   updated 应该是true？

File: agent/src/trident.rs [hunk 2]
```diff
@@ -485,7 +485,19 @@ fn dispatcher_listener_callback(
...
+                updated = false;
+            } else {
+                for i in 0..tap_types.len() {
+                    if components.cur_tap_types[i] != tap_types[i] {
+                        updated = false;
```
Comment (yuanchaoa) [inline agent/src/trident.rs:494]: 这里逻辑也不对吧

File: agent/src/trident.rs [hunk 3]
```diff
@@ -485,7 +485,19 @@ fn dispatcher_listener_callback(
...
+                    }
+                }
+            }
+            if updated {
+                components.tap_typer.on_tap_types_change(tap_types);
```
Comment (yuanchaoa) [inline agent/src/trident.rs:499]: cur_tap_types会更新么

**Pull request merged**